### ISSUE TITLE
tf.bitcast throws InternalError

### ISSUE THREAD

**Issue by trickiwoo** [tensorflow/tensorflow#59583]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.13.0.dev20230204

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 20.04.4 LTS

### Python version

3.9

### Current Behaviour?

```shell
tf.bitcast throws InternalError
```

### Standalone code to reproduce the issue

```shell
tf.bitcast(tf.cast([0,32768,65535], tf.quint16), tf.qint16)
```

### Relevant log output

```shell
tensorflow.python.framework.errors_impl.InternalError: Assigned device '/job:localhost/replica:0/task:0/device:GPU:0' does not have registered OpKernel support for _Arg
         [[{{node input}}]] [Op:Bitcast]
```
</details>

**Comment by gaikwadrahul8** [comment]
Hi, @trickiwoo 

I was able to replicate the issue on `Ubuntu 20.04` with `tf-nightly-2.13.0-dev20230204` with GPU and even I also replicated on Google Colab(`GPU`) with `tf-nightly-2.13.0-dev20230207` here is [gist file](https://colab.research.google.com/gist/gaikwadrahul8/548c2266dc8e4f53b60bd8617830e9f2/-59583-tf-nightly-gpu.ipynb) for your reference. I was getting the same error message which you mentioned in the above error log. I have added error screenshot below for your reference so it seems like we will have to dig more into this issue with `GPU` execution to find out root cause for it so we will update soon.

I replicated the same issue with `CPU` on `Ubuntu 20.04` with `tf-nightly-2.13.0-dev20230204` and on Google Colab, here is [gist file ](https://colab.research.google.com/gist/gaikwadrahul8/733b3c800824e3edfdd7546ff5556727/-59583-tf-nightly-cpu.ipynb)and it's not throwing error message `tensorflow.python.framework.errors_impl.InternalError: Assigned device '/job:localhost/replica:0/task:0/device:GPU:0' does not have registered OpKernel support for _Arg [[{{node input}}]] [Op:Bitcast] `


Here is screenshot of CPU execution :

![image](https://user-images.githubusercontent.com/115997457/217254535-1b3aa5f6-e429-4099-9818-8f361297e9e3.png)

Here is screenshot of GPU execution :

![image](https://user-images.githubusercontent.com/115997457/217256211-2cf2b65d-e20e-4533-a45d-e887d8f0cbc5.png)

Thank you for noticing the issue and we will update you soon. Thank you!
